src/pipx/shared_libs.py

```diff
@@ -73,6 +73,10 @@ def needs_upgrade(self) -> bool:
     def upgrade(
         self, *, pip_args: Optional[List[str]] = None, verbose: bool = False
     ) -> None:
+        if not self.is_valid:
```

Comment:
`is_valid` would still return True if a Windows venv’s underlying interpreter goes away. I wonder if we should actually try to execute the virtual environment instead (e.g. run `python -c "import sys; print(sys.executable)"`). This can probably be fixed separately though (when someone complains about it).

Reply:
> Also `create()` already has an `is_valid` guard.

I think that should be ok, do you see a problem?  I put this one in `upgrade` for when external code calls `upgrade` itself, not for when `create()` calls `upgrade()`.

Reply:
If you ask me, `create()` probably shouldn’t have a `is_valid` guard, since it feels weird to me calling `create()` may not actually create anything. But removing that guard probably has consequences.

Reply:
Ah that's a different matter.  I think it was meant as an optimization to save time.  We use `shared_libs.create()` every time we create a new venv.  I think probably this is meant to prevent having to recreate the shared_libs if it's already fine.

We could move the guard to the calling code in `venv.py`, which would functionally be the same thing just factored differently.  Personally I'm ok either way.

Reply:
Let’s keep this guard here and aim to remove the one in `create()` then.